source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -210,20 +212,24 @@ absl::Status ProcessorState::handleHeadersResponse(const HeadersResponse& respon
           // Do not continue filter chain here so the pending body response have chance to be
           // served.
           sendBufferedDataInStreamedMode(trailers_ == nullptr);
+          filter_.onProcessHeadersResponse(response, absl::OkStatus(), trafficDirection());
           return absl::OkStatus();
         }
       } else if (body_mode_ == ProcessingMode::BUFFERED) {
         // Here, we're not ready to continue processing because then
         // we won't be able to modify the headers any more, so do nothing and
         // let the doData callback handle body chunks until the end is reached.
         clearWatermark();
+        filter_.onProcessHeadersResponse(response, absl::OkStatus(), trafficDirection());
         return absl::OkStatus();
       } else if (body_mode_ == ProcessingMode::STREAMED) {
         sendBufferedDataInStreamedMode(false);
+        filter_.onProcessHeadersResponse(response, absl::OkStatus(), trafficDirection());
         continueIfNecessary();
         return absl::OkStatus();
       } else if (body_mode_ == ProcessingMode::FULL_DUPLEX_STREAMED) {
         // There is no buffered data in this mode.
+        filter_.onProcessHeadersResponse(response, absl::OkStatus(), trafficDirection());
```

Comment:
This indeed needs a refactor to avoid multiple place calls, but will not block this PR.

Reply:
A preexisting issue #37047 is here, would like see this is addressed at some point.